### PR TITLE
Add metadata and ping

### LINKED ISSUE #795: [0.11.1] Add Ping, MetaData, and GetMetaData to the protocol
ethereum/eth2.0-specs#1684

### PR DESCRIPTION
**Pull request by wemeetagain**
Resolves #795

### CODE REVIEW

File: packages/lodestar/src/network/reqResp.ts [hunk 1]
```diff
@@ -157,7 +168,7 @@ export class ReqResp extends (EventEmitter as IReqEventEmitterClass) implements
   }
 
   private getResponse = (peerId: PeerId, method: Method, data: Buffer): AsyncIterable<ResponseChunk> => {
-    const request = this.encoder.decodeRequest(method, data);
+    const request = method === Method.Metadata ? undefined : this.encoder.decodeRequest(method, data);
```
Comment (mpetrunic) [inline packages/lodestar/src/network/reqResp.ts:171]: Maybe we can add check formissing data in `decodeRequest`?

File: packages/lodestar/src/network/reqResp.ts [hunk 2]
```diff
@@ -148,6 +156,9 @@ export class ReqResp extends (EventEmitter as IReqEventEmitterClass) implements
     const getResponse = this.getResponse;
     return (source: AsyncIterable<Buffer|BufferList>) => {
       return (async function * () {
+        if (method === Method.Metadata) {
+          yield* getResponse(peerId, method, undefined);
+        }
```
Comment (mpetrunic) [inline packages/lodestar/src/network/reqResp.ts:161]: Maybe we should introduce better way of handling methods without response

Comment (wemeetagain) [reply packages/lodestar/src/network/reqResp.ts:161]: Definitely agree we should revisit the request/response logic to better handle methods without request / response.
I'd propose we add an issue and tackle this in a separate PR.

Comment (mpetrunic) [reply packages/lodestar/src/network/reqResp.ts:161]: :+1:

**Review by mpetrunic — APPROVED** [review]

File: packages/lodestar/src/network/reqResp.ts (hunk 2 above)
Comment (wemeetagain) [reply packages/lodestar/src/network/reqResp.ts:161]: https://github.com/ChainSafe/lodestar/issues/808

**Pull request merged**